crates/core/src/network/reducers/prune.rs

```diff
@@ -0,0 +1,32 @@
+use crate::{
+    action::{Action, ActionWrapper},
+    network::state::NetworkState,
+    state::{State, ACTION_PRUNE_MS},
+};
+use std::time::Duration;
+
+pub fn reduce_prune(
+    network_state: &mut NetworkState,
+    _root_state: &State,
+    action_wrapper: &ActionWrapper,
+) {
+    assert_eq!(action_wrapper.action(), &Action::Prune);
+
+    network_state
+        .actions
+        .iter()
+        .filter_map(|(action, response)| {
+            if let Ok(elapsed) = response.created_at.elapsed() {
+                if elapsed > Duration::from_millis(ACTION_PRUNE_MS) {
+                    return Some(action);
+                }
+            }
+            None
```

Comment:
Should we try to handle/log the error from `elapsed()` here and the other places that it is used?

Reply:
According to [the documentation](https://doc.rust-lang.org/std/time/struct.SystemTime.html), `SystemTime::elapsed()` can fail if the system clock drifts or went backwards (i.e. got (re-)set in between). I don't know how we should handle this case better than just ignoring the drift and in the worst-case waiting a bit longer. Do you have a suggestion how to handle the error?

Reply:
We might consider using `Instant` for timeout checks rather than `SystemTime`, which is less accurate, but is guaranteed to be monotonic unlike `SystemTime`. Not a big problem either way.

Reply:
BTW I pushed a commit that implements this reducer a little more straightforwardly, please review @lucksus

Reply:
Yes it also occurred to me to use [`Instant`](https://doc.rust-lang.org/std/time/struct.Instant.html), however, as you say it is less accurate, so I didn't suggest it. Alternative solutions could get complicated, and doesn't seem worth thinking about further, since the elapsed time should eventually exceed the duration for pruning.